### DISCUSSION TITLE
Redis Cluster - More clients connected on first node than on the others

### DISCUSSION THREAD

**Discussion by michelaquino** [redis/go-redis#1951] [Q&A] [answered]
Hi all,
I'm curious about a behavior of the Redis that I'm using, and I don't know if it has a relation with the library. 

I'm using a Redis cluster with 6 nodes (3 primary and 3 secondaries). This Redis is only used as a stream and I have 12 streams on it (4 in each node). The code was implemented to add messages in the streams uniformly, so the number of commands XADD  is similar to each primary node. But the number of clients and each node is different, with more clients connected on the first node.

May someone help me to understand what could be happening?

**Comment by vmihailenco** [comment, marked as answer]
Underneath, `redis.ClusterClient` uses `redis.Client` to comminucate with each node in a cluster. Each `redis.Client` maintains a separate pool of connections. Number of connections in the pool depends on number of commands that must be processed.

It is hard to tell more, but perhaps the first node receives more commands. That can happen if some keys are more popular than the others.

**Reply by michelaquino** [reply]
The number of commands is similar on each node. Does the `redis.Client` send commands to synchronize some information of cluster, like status, slots, etc? If yes, these commands are sent to all nodes or only to the first?

**Reply by vmihailenco** [reply]
It uses a random addr - https://github.com/go-redis/redis/blob/master/cluster.go#L1024

**Reply by michelaquino** [reply]
Thank you @vmihailenco for the answers
